src/workerd/api/r2-bucket.c++

```diff
@@ -521,11 +531,22 @@ R2Bucket::put(jsg::Lock& js, kj::String name, kj::Maybe<R2PutValue> value,
     }
 
     auto requestJson = json.encode(requestBuilder);
+    auto account = adminAccount.map([](auto&& s) { return kj::str(s); });
     auto bucket = adminBucket.map([](auto&& s) { return kj::str(s); });
 
+    kj::Vector<kj::StringPtr> path;
+    KJ_IF_MAYBE(a, account) {
+      path.add(*a);
+      // TODO throw assertion if account is defined but not bucket
```

Comment:
Need to throw an assertion in here, since bucket needs to also be set. Not sure how to write the `KJ_ASSERT` for it and am open to suggestions.

Reply:
There's KJ_ASSERT_NONNULL, does that help?